validator/client/redisproducer.go

```diff
@@ -0,0 +1,130 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/offchainlabs/nitro/pubsub"
+	"github.com/offchainlabs/nitro/util/containers"
+	"github.com/offchainlabs/nitro/util/redisutil"
+	"github.com/offchainlabs/nitro/util/stopwaiter"
+	"github.com/offchainlabs/nitro/validator"
+	"github.com/offchainlabs/nitro/validator/server_api"
+	"github.com/offchainlabs/nitro/validator/server_common"
+	"github.com/spf13/pflag"
+)
+
+type RedisValidationClientConfig struct {
+	Name           string                `koanf:"name"`
+	Room           int32                 `koanf:"room"`
+	RedisURL       string                `koanf:"redis-url"`
+	ProducerConfig pubsub.ProducerConfig `koanf:"producer-config"`
+	// Supported wasm module roots, when the list is empty this is disabled.
+	ModuleRoots []string `koanf:"module-roots"`
```

Comment:
add a moduleRoots private member, which is a []common.Hash
add a Validate() function to RedisValidationClientConfig

Validate function will parse the strings into hashes and return error if there is one
Call this Validate from it's parent config's Validate
then inside the code use the parsed member.